### PR TITLE
feat: add AAP backend plugin

### PR DESCRIPTION
**Pull request by invincibleJai**
## Description

Adds AAP (Ansible automation platform) backend plugin to the Showcase App.

## Which issue(s) does this PR fix

- Fixes https://github.com/janus-idp/backstage-showcase/issues/496

## Screenshots

image

image

image

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [x] Documentation is updated if necessary (requirement for new features)
- [x] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Comment by Zaperex** [comment]
I think you'll need to add a changeset for the backend package with `yarn changeset`

**Review by christophe-f — CHANGES_REQUESTED** [review]

**Comment by christophe-f** [comment]
The plugin can be added to the list of plugins on the README

File: app-config.yaml [hunk 1]
```diff
@@ -223,6 +223,14 @@ catalog:
     #   providerId:
     #     organization: "${GITHUB_ORG}"
 
+    # Plugin: AAP backend plugin
+    aap:
+      prod:
+        baseUrl: ${AAP_BASE_URL}
+        authorization: ${AAP_AUTH_TOKEN}
+        owner: janus-authors
+        system: janus-idp
```
[the remark this answers is not in the document]
Comment (Zaperex) [reply app-config.yaml:233]: I believe the way the AAP backend plugin currently works is that it grabs job/workflow templates from AAP (so there's no `catalog-info.yaml` to refer to), which is why it's grabbing the information for the owner and system from the `app-config.yaml` to create the AAP catalog entities

**Comment by invincibleJai** [comment]
> The plugin can be added to the list of plugins on the README

updated, @christophe-f PTAL when you get time.

File: app-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (invincibleJai) [reply app-config.yaml:233]: yeah, thanks @Zaperex for the comment.

File: packages/backend/package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "@immobiliarelabs/backstage-plugin-gitlab-backend": "6.2.0",
     "@janus-idp/backstage-plugin-keycloak-backend": "1.5.3",
     "@janus-idp/backstage-plugin-ocm-backend": "3.2.1",
+    "@janus-idp/backstage-plugin-aap-backend": "^1.2.2",
```
Comment (Zaperex) [inline packages/backend/package.json:48]: I think we should pin the dependency similar to how we're dealing with all the other dependencies

File: packages/backend/src/plugins/catalog.ts [hunk 3]
```diff
@@ -106,6 +108,18 @@ export default async function createPlugin(
     );
   }
 
+  if (isAapEnabled) {
+    builder.addEntityProvider(
+      AapResourceEntityProvider.fromConfig(env.config, {
+        logger: env.logger,
+        schedule: env.scheduler.createScheduledTaskRunner({
+          frequency: { minutes: 1 },
```
Comment (Zaperex) [inline packages/backend/src/plugins/catalog.ts:116]: All the other Entity Providers installed in the showcase currently use a frequency of 30 minutes or 1 hour, I think 1 minute might be a bit too frequent?

File: app-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (christophe-f) [reply app-config.yaml:233]: ok, then is it something we can remove?

[the remark this answers is not in the document]
Comment (invincibleJai) [reply app-config.yaml:244]: yeah, we can remove it as it's optional. If not provided it'll not show it and in OCM config we do pass owner https://github.com/janus-idp/backstage-showcase/blob/main/app-config.yaml#L224

[the remark this answers is not in the document]
Comment (christophe-f) [reply app-config.yaml]: yes it's good now. I guess we will have to remove this from the OCM plugin, but it's not related to this PR.

**Review by christophe-f — APPROVED** [review]
LGTM

**Pull request merged**